x-pack/plugins/lens/public/embeddable/embeddable.tsx

```diff
@@ -260,6 +269,26 @@ export class Embeddable
     return this.lensInspector.adapters;
   }
 
+  private maybeAddConflictError (errors: ErrorMessage[], sharingSavedObjectProps?: SharingSavedObjectProps) {
+    const ret = [...errors];
+
+    if (sharingSavedObjectProps?.outcome === 'conflict' && !!this.deps.spaces) {
+      ret.push({
+        shortMessage: i18n.translate('xpack.lens.embeddable.legacyURLConflict.shortMessage', {
+          defaultMessage: `You've encountered a URL conflict`,
+        }),
+        longMessage: (
+          <this.deps.spaces.ui.components.getEmbeddableLegacyUrlConflict
+            targetType={DOC_TYPE}
+            sourceId={sharingSavedObjectProps.sourceId!}
+          />
+        ),
+      })
+    }
+
+    return ret;
+  }
```

Comment:
I feel like pulling this out into its own method made the `initializeSavedVis` more readable, but this is optional.